### PR TITLE
output can take a username as an arg

### PR DESCRIPTION
**Pull request by aerickson**
Output entries starting with U will be converted to IM channels.

Please let me know if there's a better way to go about this.

Thanks,
Andy

### CODE REVIEW

File: rtmbot.py [hunk 1]
```diff
@@ -55,12 +55,19 @@ def output(self):
         for plugin in self.bot_plugins:
             limiter = False
             for output in plugin.do_output():
-                channel = self.slack_client.server.channels.find(output[0])
-                if channel != None and output[1] != None:
+                destination = output[0]
+                message = output[1]
+                # things that start with U are users. convert to an IM channel.
+                if destination.startswith('U'):
+                    result = json.loads(self.slack_client.api_call('im.open', user=destination))
```
Comment (Yasumoto) [inline rtmbot.py]: mind doing a `try/except` to catch `ValueError` in case there's some incorrectly formatted JSON returned?

File: rtmbot.py [hunk 2]
```diff
@@ -55,12 +55,19 @@ def output(self):
...
+                    channel = self.slack_client.server.channels.find(result[u'channel'][u'id'])
```
Comment (Yasumoto) [inline rtmbot.py]: These would benefit from  either a `result.get(u'channel', {}).get(u'id', None)` or a `try/except` for `KeyError`

File: rtmbot.py [hunk 3]
```diff
@@ -55,12 +55,19 @@ def output(self):
         for plugin in self.bot_plugins:
             limiter = False
             for output in plugin.do_output():
-                channel = self.slack_client.server.channels.find(output[0])
-                if channel != None and output[1] != None:
+                destination = output[0]
+                message = output[1]
+                # things that start with U are users. convert to an IM channel.
+                if destination.startswith('U'):
```
Comment (Yasumoto) [inline rtmbot.py]: Mind also checking for `'G'` (which is a [Group Message](https://api.slack.com/methods/groups.open))?

```
elif destination.startswith('G'):
    result = self.slack_client.api_call('groups.open'), channel=destination)
#...
```

**Comment by Yasumoto** [comment]
This is a great idea! If you're still interested in this, mind rebasing and adding some tweaks? Looking forward to getting this in!

**Comment by aerickson** [comment]
@Yasumoto Sorry for the delay. I'm not exactly sure how the group messaging should work. Please let me know if you think of anything else.

**Pull request merged**